package/MDAnalysis/core/topologyattrs.py

```diff
@@ -953,15 +958,23 @@ def radius_of_gyration(group, **kwargs):
 
 
         .. versionchanged:: 0.8 Added *pbc* keyword
+        .. versionchanged:: 0.20.0 Added *unwrap* and *compound* parameter
 
         """
         atomgroup = group.atoms
         pbc = kwargs.pop('pbc', flags['use_pbc'])
         masses = atomgroup.masses
+        unwrap = kwargs.pop('unwrap', False)
+        compound = kwargs.pop('compound', 'group')
+
+        com = atomgroup.center_of_mass(pbc=pbc, unwrap=unwrap, compound=compound)
+        if compound is not 'group':
+            com = (com * group.masses[:, None]).sum(axis=0) / group.masses.sum()
```

Comment:
You've added a compound parameter and not written tests for it, so this line never gets tested.  What is this line doing?

Reply:
it gets tested [here]( https://github.com/MDAnalysis/mdanalysis/pull/2299/files#diff-ab92738dd8fc8e62b290ddbdd5eb909dR912)

If the compound is 'residue', atomgroup.center_of_mass will return COM of each residue. This line takes the mean of each COM.

Reply:
@richardjgowers is this sufficient or should I add some more tests?